arm/Microsoft.DomainServices/AzureADDomainServices/readme.md

```diff
@@ -0,0 +1,160 @@
+| `filteredSync` | string | `enabled`  |  | Optional: The value is to synchronise scoped users and groups. |
+| `tlsV1` | string | `enabled`  |  | Optional: The value is to enable clients making request using TLSv1. |
+| `ntlmV1` | string | `enabled` | | Optional: The value is to enable clients making request using NTLM v1. |
+| `syncNtlmPasswords` | string  | `enabled` |  | Optional: The value is to enable synchronised users to use NTLM authentication. |
+| `syncOnPremPasswords` | string | `enabled`  |  | Optional: The value is to enable on-premises users to authenticate against managed domain. |
+| `kerberosRc4Encryption` | string | `enabled`  |  | Optional: The value is to enable Kerberos requests that use RC4 encryption. |
+| `kerberosArmoring` | string | `enabled`  |  | Optional: The value is to enable to provide a protected channel between the Kerberos client and the KDC. |
+| `notifyDcAdmins` | string  | `enabled`  |  | Optional: The value is to notify the DC Admins. |
+| `notifyGlobalAdmins` | string | `enabled` |  | Optional: The value is to notify the Global Admins. |
+| `ldapexternalaccess` | string  | `enabled`  |  | Required: The value is to enable the Secure LDAP for external services of Azure ADDS Services. |
+| `secureldap` | string  | `enabled`  |  | Required: The value is to enable the Secure LDAP for Azure ADDS Services.|
+| `diagnosticStorageAccountId ` | string  |  |  | Optional. Resource ID of the diagnostic storage account. |
+| `diagnosticWorkspaceId ` | string  |  |  | Optional. Resource ID of the diagnostic log analytics workspace. |
+| `diagnosticEventHubAuthorizationRuleId ` | string  |  |  | Optional. Resource ID of the diagnostic event hub authorization rule for the Event Hubs namespace in which the event hub should be created or streamed to. |
+| `diagnosticEventHubName ` | string  | |  | Optional. Name of the diagnostic event hub within the namespace to which logs are streamed. Without this, an event hub is created for each log category. |
+| `tags ` | object | `{object}` |  | Optional. Tags of the resource. |
+| `diagnosticLogsRetentionInDays` | int  | 365 |  | Optional. Specifies the number of days that logs will be kept for; a value of 0 will retain data indefinitely. |
+| `lock  ` | string |  |  | Optional. Specify the type of lock. |
+| `roleAssignments  ` | array  | `[array]` |  | Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\' |
+| `logsToEnable   ` | array  | `[array]` |  | optional. The name of logs that will be streamed. |
+
+### Parameter Usage: `Azure ADDS`
```

Comment:
I'd suggest to move this into a dedicated header like `## Examples` with one subheader `### Example 1` as the `Parameter Usage` is intended specifically to explain complex single parameters such as arrays / objects

Reply:
There is already a dedicated section called roleAssignments in the readme file or do you wish me to break this further?

Reply:
Correct, as `tags` & `roleAssignments` are complex objects/arrays, it makes sense to explain their construction outhside of the parameters table in more detail. However, note that both are properties of the module.

The example you here is instead a parameter file for the entire module deployment (which I guess maps 1:1 to the `aadds.parameter.json` file).

I hence see 2 possible actions:
- Either we remove this example as it is already documented in the `aadds.parameter.json` file, or
- We move it into its own section such as `## Deployment Example` below the `## Parameters` section.

Reply:
@MrMCake yes, I agree that the content in aadds.parameters.json is same(duplicate) as in readme.md file. I am happy with the option-1 if you are fine with it. I can go ahead and remove the section "Parameter Usage: Azure ADDS" from readme.md file then.

Reply:
Hey @mararam, excuse the late reply. Sounds good, please do :)

Reply:
@MrMCake I have deleted the section now. Thank you.